platforms/android/src/jp/wizcorp/phonegap/plugin/WizAssets/HttpToFile.java

```diff
@@ -0,0 +1,132 @@
+package jp.wizcorp.phonegap.plugin.WizAssets;
+import java.io.*;
+import java.net.Authenticator;
+import java.net.HttpURLConnection;
+import java.net.PasswordAuthentication;
+import java.net.URL;
+
+public final class HttpToFile {
+    private static int _blockSize;
+    private static final String TAG = "WizAssetsPlugin";
+    private static ILogger _logger;
+
+    public static void setLogger(ILogger logger) {
+        _logger = logger;
+    }
+
+    public static void setBlockSize(int blockSize) {
+        _blockSize = blockSize;
+    }
+
+    public static int downloadFile(URL url, File file) throws IOException, Exception {
+        _logger.logDebug(TAG, "[Downloading to] " + file.getAbsolutePath());
+        BufferedInputStream inputStream = null;
+        HttpURLConnection urlConnection = null;
+        int httpStatus = -1;
+        try {
+            if (!createPath(file)) {
+                _logger.logError(TAG, "file path error");
+            } else {
+                authenticate(url);
+
+                urlConnection = (HttpURLConnection) url.openConnection();
+                urlConnection = handleRedirect(urlConnection);
+                httpStatus = urlConnection.getResponseCode();
+                inputStream = new BufferedInputStream(urlConnection.getInputStream());
+                writeFile(inputStream, file);
+            }
+        } catch (Exception e) {
+            printError(e);
+            throw e;
+        } finally {
+            try {
+                closeStream(inputStream);
+            } finally {
+                if (urlConnection != null) {
+                    urlConnection.disconnect();
+                }
+            }
+        }
+        return httpStatus;
+    }
+
+    private static HttpURLConnection handleRedirect(HttpURLConnection urlConnection) throws IOException {
+        int status = urlConnection.getResponseCode();
+        while (status != HttpURLConnection.HTTP_OK
```

Comment:
Since the status will usually be HttpURLConnection.HTTP_OK, it doesn't have to check the other values.  A small optimization.  So I would argue that it is needed.